get-latest-version.py

```diff
@@ -3,196 +3,164 @@
-        ],
-        check=True,
-    )
-    # Push the new branch to the remote repository
-    subprocess.run(["git", "push", "origin", new_branch], check=True)
-    pr = repo.create_pull(
-        base=base_branch,
-        head=new_branch,
-        title=pull_title,
-        body=f"""A new version of `{component_name} {pocket_version}` replaces `{patched_version}`.""",
-    )
-    subprocess.run(["git", "switch", "master"], check=True)
+    for package_and_upstream in packages_and_upstream:
+        package_name, *upstream_series_name = package_and_upstream.split(":", 1)
+        upstream_series_name = (
+            upstream_series_name[0] if upstream_series_name else series_name
+        )
+        print(package_name, upstream_series_name)
+
+        series = ubuntu.getSeries(name_or_version=series_name)
+        upstream_series = ubuntu.getSeries(name_or_version=upstream_series_name)
+
+        patched_sources = patches_archive.getPublishedSources(
+            exact_match=True,
+            source_name=package_name,
+            status="Published",
+            distro_series=series,
+        )
+        patched_version = patched_sources[0].source_package_version
+
+        # Search for a new version in the Ubuntu repositories
+        for pocket in ["Release", "Security", "Updates"]:
+
+            upstream_sources = ubuntu_archive.getPublishedSources(
+                exact_match=True,
+                source_name=package_name,
+                status="Published",
+                pocket=pocket,
+                distro_series=upstream_series,
+            )
+
+            if len(upstream_sources) <= 0:
+                continue
+
+            pocket_version = upstream_sources[0].source_package_version
+            if apt_pkg.version_compare(pocket_version, patched_version) <= 0:
+                continue
+
+            pull_title = f"📦 Update {package_name} [{upstream_series_name}]"
+            if github_pull_exists(pull_title, repo):
+                continue
+
+            base_branch = f"{package_name}-{upstream_series_name}"
+            new_branch = f"bot/update/{package_name}-{upstream_series_name}"
+
+            # Checkout the base branch
+            current_repo.git.checkout(base_branch)
+            # Create and checkout the new branch
+            current_repo.git.checkout("-b", new_branch)
+
+            p_apt_source = subprocess.run(
+                f"apt source {package_name}",
+                shell=True,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                check=True,
+                encoding="utf-8",
+            )
+
+            # getting the directory to which the source is extracted
+            extraction_dest = ""
+            for line in p_apt_source.stdout.split("\n"):
+                if "dpkg-source: info: extracting" in line:
+                    print(line)
+                    extraction_dest = line.split()[-1]
```

Comment:
I don't think this suggestion is better, but I thought we can simply use `find` too here like the following example, because there should be no other directories exist when we're here.

```
user@elementary-8:~/work$ find . -maxdepth 1 -mindepth 1 -type d
./io.elementary.camera-6.2.0+pkg23~ubuntu8.1
user@elementary-8:~/work$
```

Reply:
Ah OK then my suggestion wouldn't work here.